vitrina/datasets/views.py

```diff
@@ -42,3 +47,50 @@ def get_context_data(self, **kwargs):
         }
         context_data.update(extra_context_data)
         return context_data
+
+
+class DatasetCreateView(CreateView):
+    model = Dataset
+    template_name = 'base_form.html'
+    context_object_name = 'dataset'
+    form_class = NewDatasetForm
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated and\
+                request.user.organization is not None and\
+                request.user.organization.slug == self.kwargs['slug']:
+            return super(DatasetCreateView, self).get(request, *args, **kwargs)
+        else:
+            return redirect('organization-detail', self.kwargs['org_kind'], self.kwargs['slug'])
```

Comment:
```suggestion
            return redirect(org)
```

And before that, we should get the org `get_object_or_404(Organization, ...)`. We can't trust user input.